### PR TITLE
deps: update buildkit, vendor changes

### PR DESCRIPTION
**Pull request by laurazard**
PTAL, they're small changes but I don't really have the context to make sure this is correct.

Related to https://github.com/docker/compose/pull/11283, doing these changes would enable bumping Buildx and Buildkit in Compose to master and solve the OTEL trouble blocking that PR

### CODE REVIEW

**Comment by thaJeztah** [comment]
Let me also check if https://github.com/moby/buildkit/pull/4496 is good to go

File: go.mod [hunk 1]
```diff
@@ -24,7 +24,7 @@ require (
 	github.com/google/uuid v1.3.0
 	github.com/hashicorp/go-cty-funcs v0.0.0-20230405223818-a090f58aa992
 	github.com/hashicorp/hcl/v2 v2.19.1
-	github.com/moby/buildkit v0.13.0-beta1.0.20231129145300-c9ee8491d74f // master (v0.13.0-dev)
+	github.com/moby/buildkit v0.13.0-beta1.0.20231218103508-bbb6aaccb9ae // master (v0.13.0-dev)
```
Comment (crazy-max) [inline go.mod:27]: Can you vendor latest changes so it brings https://github.com/moby/buildkit/pull/4496?

Comment (laurazard) [reply go.mod:27]: will do

**Comment by laurazard** [comment]
PTAL @crazy-max

**Review by crazy-max — APPROVED** [review]
LGTM, the metrics exporter introduced by https://github.com/moby/buildkit/pull/4460 will be handled in follow-up (cc @jsternberg)

**Pull request merged**